src/Adapter/Order/CommandHandler/AddProductToOrderHandler.php

```diff
@@ -89,132 +99,155 @@ public function handle(AddProductToOrderCommand $command)
-            $orderCarrier->weight = (float) $order->getTotalWeight();
-            if ($orderCarrier->update()) {
-                $order->weight = sprintf('%.3f ' . Configuration::get('PS_WEIGHT_UNIT'), $orderCarrier->weight);
+            $precision = $this->getPrecisionFromCart($cart);
+            $order->total_paid = Tools::ps_round(
+                (float) $cart->getOrderTotal(true, $totalMethod, $orderProducts),
+                $precision
+            );
+            $order->total_paid_tax_excl = Tools::ps_round(
+                (float) $cart->getOrderTotal(false, $totalMethod, $orderProducts),
+                $precision
+            );
+            $order->total_paid_tax_incl = Tools::ps_round(
+                (float) $cart->getOrderTotal(true, $totalMethod, $orderProducts),
+                $precision
+            );
+            $order->total_shipping = $cart->getOrderTotal(true, Cart::ONLY_SHIPPING, $orderProducts);
+            $order->total_shipping_tax_excl = $cart->getOrderTotal(false, Cart::ONLY_SHIPPING, $orderProducts);
+            $order->total_shipping_tax_incl = $cart->getOrderTotal(true, Cart::ONLY_SHIPPING, $orderProducts);
+
+            $order->total_wrapping = abs($cart->getOrderTotal(true, Cart::ONLY_WRAPPING, $orderProducts));
+            $order->total_wrapping_tax_excl = abs($cart->getOrderTotal(false, Cart::ONLY_WRAPPING, $orderProducts));
+            $order->total_wrapping_tax_incl = abs($cart->getOrderTotal(true, Cart::ONLY_WRAPPING, $orderProducts));
+
+            StockAvailable::synchronize($product->id);
+
+            // Update weight SUM
+            $orderCarrier = new OrderCarrier((int) $order->getIdOrderCarrier());
+            if (Validate::isLoadedObject($orderCarrier)) {
+                $orderCarrier->weight = (float) $order->getTotalWeight();
+                if ($orderCarrier->update()) {
+                    $order->weight = sprintf('%.3f ' . Configuration::get('PS_WEIGHT_UNIT'), $orderCarrier->weight);
+                }
             }
-        }
 
-        // Update Tax lines
-        $orderDetail->updateTaxAmount($order);
+            // Update Tax lines
+            $orderDetail->updateTaxAmount($order);
 
-        // Delete specific price if exists
-        if (null !== $specificPrice) {
-            $specificPrice->delete();
-        }
+            // Delete specific price if exists
+            if (null !== $specificPrice) {
+                $specificPrice->delete();
+            }
 
-        $order = $order->refreshShippingCost();
+            $order = $order->refreshShippingCost();
 
-        Hook::exec('actionOrderEdited', ['order' => $order]);
+            Hook::exec('actionOrderEdited', ['order' => $order]);
 
-        $newCartRules = $this->context->cart->getCartRules();
+            $newCartRules = $this->context->cart->getCartRules();
 
-        sort($oldCartRules);
-        sort($newCartRules);
+            sort($oldCartRules);
```

Comment:
Since old and new cart rules now come from the same cart instance, `$newCartRules` are only the ones added from the product, maybe this could be fetched more easily